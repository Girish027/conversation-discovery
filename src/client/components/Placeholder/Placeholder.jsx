import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Placeholder extends PureComponent {
  render() {
    const {
      children, styleOverride, ...otherProps
    } = this.props;
    if (children) {
      return (
        <div
          className='vertical-center horizontal-center'
          style={{
            paddingBottom: '20px',
            ...styleOverride
          }}
          {...otherProps}
        >
          {children}
        </div>
      );
    }
    return '';
  }
}

Placeholder.defaultProps = {
  styleOverride: {},
};

Placeholder.propTypes = {
  children: PropTypes.node,
  styleOverride: PropTypes.object,
};

export default Placeholder;
