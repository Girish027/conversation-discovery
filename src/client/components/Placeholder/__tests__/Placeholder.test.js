import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Placeholder from 'components/Placeholder';

describe('<Placeholder />', () => {
  let wrapper;

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = shallow(<Placeholder />);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly with given information', () => {
      wrapper = shallow(
        <Placeholder>
          placeholder message
        </Placeholder>
      );
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
