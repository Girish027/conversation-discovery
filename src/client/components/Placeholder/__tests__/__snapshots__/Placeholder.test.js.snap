// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Placeholder /> Snapshots renders correctly with given information 1`] = `
<div
  className="vertical-center horizontal-center"
  style={
    Object {
      "paddingBottom": "20px",
    }
  }
>
  placeholder message
</div>
`;
